### PR TITLE
DEVEXP-987: remove support of default region

### PR DESCRIPTION
**Pull request by JPPortier**

### CODE REVIEW

File: client/src/main/com/sinch/sdk/domains/conversation/api/v1/adapters/ConversationService.java [hunk 1]
```diff
@@ -249,20 +249,13 @@ private void instanceLazyInit() {
             credentials.getKeySecret(), "Conversation service requires 'keySecret' to be defined");
         StringUtil.requireNonEmpty(
             credentials.getProjectId(), "Conversation service requires 'projectId' to be defined");
+        Objects.requireNonNull(
+            context.getRegion(), "Conversation service requires 'region' to be defined");
+        StringUtil.requireNonEmpty(
```
Comment (rpredescu-sinch) [inline client/src/main/com/sinch/sdk/domains/conversation/api/v1/adapters/ConversationService.java:254]: Hi @JPPortier!

Another idea would be to add `StringUtil.requireNonEmpty` in `ConversationRegion.from` and remove it from here.

```
public static ConversationRegion from(String value) {
    StringUtil.requireNonEmpty(value, "ConversationRegion value cannot be empty");
    return ENUM_SUPPORT.from(value);
}
```

This way we do not have to add the string validation anywhere else. Same for `SMSService`.

Best!

File: client/src/test/java/com/sinch/sdk/domains/conversation/api/v1/adapters/ConversationServiceTest.java [hunk 2]
```diff
@@ -68,143 +55,13 @@ void checkCredentialsTranscoding() {
...
-
-    Optional<LogRecord> record =
-        logRecorder.records.stream()
-            .filter(
-                r -> r.getMessage().contains("Using default region for Conversation Template 'us"))
-            .findFirst();
-
-    assertFalse(record.isPresent());
-    LogManager.getLogManager().reset();
+  void passInitWithoutSettings() {
```
Comment (rpredescu-sinch) [inline client/src/test/java/com/sinch/sdk/domains/conversation/api/v1/adapters/ConversationServiceTest.java:63]: Hi, @JPPortier!

Could be better to have a dedicated TemplatesServiceTest.java with this test (and any future template-specific tests)? Nice to have. 

Best!

File: client/src/test/java/com/sinch/sdk/domains/sms/api/v1/adapters/CredentialsValidationHelper.java
Comment (rpredescu-sinch) [inline client/src/test/java/com/sinch/sdk/domains/sms/api/v1/adapters/CredentialsValidationHelper.java:1]: Hi @JPPortier !
This file name do not have the `Test` suffix as the others do.
Best!

File: client/src/main/com/sinch/sdk/domains/conversation/templates/api/adapters/TemplatesService.java [hunk 3]
```diff
@@ -82,22 +82,12 @@ private void instanceLazyInit() {
             credentials.getKeySecret(), "Templates service requires 'keySecret' to be defined");
         StringUtil.requireNonEmpty(
             credentials.getProjectId(), "Templates service requires 'projectId' to be defined");
-
+        Objects.requireNonNull(
```
Comment (rpredescu-sinch) [inline client/src/main/com/sinch/sdk/domains/conversation/templates/api/adapters/TemplatesService.java:85]: Hi, @JPPortier!
Only `requireNonNull(region)` was added. Do we also need `requireNonEmpty(region.value())`?
Best!

File: client/src/test/java/com/sinch/sdk/domains/conversation/api/templates/adapters/CredentialsValidationHelper.java [hunk 4]
```diff
@@ -147,6 +151,26 @@ private static void doNotAcceptNullURLTemplates(
             .contains("Templates service requires 'templateManagementUrl' to be defined"));
   }
 
+  private static void doNotAcceptNullRegionTemplates(
```
Comment (rpredescu-sinch) [inline client/src/test/java/com/sinch/sdk/domains/conversation/api/templates/adapters/CredentialsValidationHelper.java:155]: Hi @JPPortier !
This test only covers the null region case. Should we also have a test for empty region `ConversationRegion.from("")`?
Best!

File: client/src/main/com/sinch/sdk/domains/conversation/api/v1/adapters/ConversationService.java (hunk 1 above)
Comment (JPPortier) [reply client/src/main/com/sinch/sdk/domains/conversation/api/v1/adapters/ConversationService.java:254]: Good point.
The general pattern of dynamic enums here across all of the SDK is to accept any new value without any dedicated business logic (same pattern applied for values coming from backend for example onto responses)

Then by accepting an empty value we can have dedicated business logic when it is required for usage (here during init of service which is checking it own requirements).

If another location (e.g. SDK user sources level) do not have same requirements, an empty value can be accepted without issue.

So in my opinion, it is better to let business logic validation where it have more sense: on use

File: client/src/main/com/sinch/sdk/domains/conversation/templates/api/adapters/TemplatesService.java (hunk 3 above)
Comment (JPPortier) [reply client/src/main/com/sinch/sdk/domains/conversation/templates/api/adapters/TemplatesService.java:85]: 👌 
Fixed

File: client/src/test/java/com/sinch/sdk/domains/conversation/api/templates/adapters/CredentialsValidationHelper.java (hunk 4 above)
Comment (JPPortier) [reply client/src/test/java/com/sinch/sdk/domains/conversation/api/templates/adapters/CredentialsValidationHelper.java:155]: Good catch 👌 
added

File: client/src/test/java/com/sinch/sdk/domains/conversation/api/v1/adapters/ConversationServiceTest.java (hunk 2 above)
Comment (JPPortier) [reply client/src/test/java/com/sinch/sdk/domains/conversation/api/v1/adapters/ConversationServiceTest.java:63]: 🤦 
cut/paste issue.
Test is really for ConversationService (related Templates one  already exist onto TemplateServiceTest)

Updated

File: client/src/test/java/com/sinch/sdk/domains/sms/api/v1/adapters/CredentialsValidationHelper.java
Comment (JPPortier) [reply client/src/test/java/com/sinch/sdk/domains/sms/api/v1/adapters/CredentialsValidationHelper.java:1]: Yes.

Like other CredentialsValidationHelpers files (SMS, Conversation, ConversationTemplate): it do not contains any test declaration, just helpers called by them

**Review by asein-sinch — APPROVED** [review]
